server/cluster_worker.go

```diff
@@ -113,3 +115,41 @@ func (c *RaftCluster) handleAskSplit(request *pdpb.AskSplitRequest) (*pdpb.AskSp
 
 	return split, nil
 }
+
+func (c *RaftCluster) checkSplitRegion(left *metapb.Region, right *metapb.Region) error {
+	if left == nil || right == nil {
+		return errors.New("invalid split region")
+	}
+
+	if !bytes.Equal(left.GetEndKey(), right.GetStartKey()) {
+		return errors.New("invalid split region")
+	}
+
+	if bytes.Compare(left.GetStartKey(), right.GetEndKey()) >= 0 {
+		return errors.New("invalid split region")
+	}
+
+	return nil
+}
+
+func (c *RaftCluster) handleReportSplit(request *pdpb.ReportSplitRequest) (*pdpb.ReportSplitResponse, error) {
+	left := request.GetLeft()
+	right := request.GetRight()
+
+	err := c.checkSplitRegion(left, right)
+	if err != nil {
+		log.Warnf("report split region is invalid - %v, %v", request, errors.ErrorStack(err))
+		return nil, errors.Trace(err)
+	}
+
+	// Build origin region by using left and right.
+	originRegion := cloneRegion(left)
+	originRegion.RegionEpoch = nil
+	originRegion.EndKey = right.GetEndKey()
+
+	// Wrap report split message as an Operator, and added into cache.
+	op := newSplitOperator(originRegion, left, right)
+	c.balancerWorker.historyOperators.add(originRegion.GetId(), op)
+
+	return &pdpb.ReportSplitResponse{}, nil
```

Comment:
Yes, it is only a report from tikv, like store/region heartbeat, there is no need to return a detail response.